### DISCUSSION TITLE
How to specify config dir path?

### DISCUSSION THREAD

**Discussion by greensea** [milvus-io/milvus#10235] [Q&A and General discussion] [answered]
I compile milvus from source, then I start it with
```
milvus run standalone
```
milvus tells me it is loading config files from `/home/gs/dl/milvus-2.0.0-testing-20211018/configs/`, this is the build dir. How can I ask milvus to use another config dir?

**Comment by yhmo** [comment, marked as answer]
This is the function of how milvus find the config path:
```
func (gp *BaseTable) initConfPath() string {
	// check if user set conf dir through env
	configDir, find := syscall.Getenv("MILVUSCONF")
	if !find {
		runPath, err := os.Getwd()
		if err != nil {
			panic(err)
		}
		configDir = runPath + "/configs/"
		if _, err := os.Stat(configDir); err != nil {
			_, fpath, _, _ := runtime.Caller(0)
			// TODO, this is a hack, need to find better solution for relative path
			configDir = path.Dir(fpath) + "/../../../configs/"
		}
	}
	return configDir
}
```
So, we can use environment variable "MILVUSCONF" to change the config path, otherwise it will use "/configs/" under execute path as config path.

**Reply by greensea** [reply]
Thanks.
I think the docs should list all available ENVs, I was struggling searching everywhere but didn't find this ENV name.

**Reply by yhmo** [reply]
Right, this is a good suggestion. I just create an issue about this: https://github.com/milvus-io/milvus/issues/10290
